src/main/java/Reporter.java

```diff
@@ -27,65 +28,99 @@
   public static void main(String [] argv) {
     Reporter reporter = new Reporter();
     try {
-      Future<HttpClientResponse> fut1 = Future.future();
-      Future<HttpClientResponse> fut2 = Future.future();
-      Future<HttpClientResponse> fut3 = Future.future();
-      Future<HttpClientResponse> fut4 = Future.future();
-      Future<HttpClientResponse> fut5 = Future.future();
-
-      HttpClientRequest request = reporter.getHttpRequest(host, feedURI, HttpMethod.POST).handler(response -> {
-        if (response.statusCode() == 201) {
+      Future<Void> fut0 = Future.future();
+      Future<Void> fut1 = Future.future();
+      Future<Void> fut2 = Future.future();
+      Future<Void> fut3 = Future.future();
+      Future<Void> fut4 = Future.future();
+      Future<Void> fut5 = Future.future();
+      // step 0: create feed
+      HttpClientRequest request = reporter.getHttpRequest(host, feedURI, HttpMethod.POST).handler(resp -> {
+        if (resp.statusCode() == 201) {
           System.out.println("Created feed!");
-          fut1.complete();
+          fut0.complete();
         } else {
-          fut1.fail("Fail when Creating feed");
+          fut0.fail("Fail when creating feed");
         }
       });
-      JsonObject json = new JsonObject().put("id", feedId);
-      request.end(json.encode());
-
-      fut1.compose(response -> {
-        // step 2: create a vertx
-        HttpClientRequest request1 = reporter.getHttpRequest(host, resourceURI, HttpMethod.POST).handler(response1 -> {
-          System.out.println("Created vertx resource!");
-          fut2.complete();
+      request.end(new JsonObject().put("id", feedId).encode());
+
+      fut0.compose(Void -> {
+        // step 1: create customized resource type
+        HttpClientRequest req = reporter.getHttpRequest(host, resourceTypeURI, HttpMethod.POST).handler(resp -> {
+          if (resp.statusCode() == 201) {
+            System.out.println("Created customized resource type");
+            fut1.complete();
+          } else {
+            fut1.fail("Fail when creating resource type");
+          }
         });
-        JsonObject json1 = new JsonObject().put("id", vertxResourceId).put("resourceTypePath", "/rt;URL")
+        JsonObject json = new JsonObject().put("id", "MYRT");
+        req.end(json.encode());
+      }, fut1);
+
+      fut1.compose(Void -> {
+        // step 2: create a vertx resource
+        HttpClientRequest req = reporter.getHttpRequest(host, resourceURI, HttpMethod.POST).handler(resp -> {
+          if (resp.statusCode() == 201) {
+            System.out.println("Created vertx resource!");
+            fut2.complete();
+          } else {
+            fut2.fail("Fail when creating resource vertx");
+          }
+        });
+        JsonObject json = new JsonObject().put("id", vertxResourceId).put("resourceTypePath", "/f;" + feedId + "/rt;MYRT")
                 .put("properties", new JsonObject().put("type", "standalone"));
```

Comment:
So this is how you add a Vert.x resource, declaring it is of type MYRT, correct? 

Is there a reason the 'type' property of you Vert.x resource is not declared in the resource type?